lib/TPP/DefaultPipeline.cpp

```diff
@@ -124,8 +145,16 @@ struct DefaultPipeline : public tpp::impl::DefaultPipelineBase<DefaultPipeline>,
       pm.addPass(createGpuPipeline(GpuPipelineOptions{gpuBackend}));
     } else {
       // Apply the default preprocessing pass
-      DefaultTppPassesOptions tppDefaultOptions{linalgToLoops,
-                                                parallelTaskGrid};
+      DefaultTppPassesOptions tppDefaultOptions;
+      if (!tileShapeM.empty())
```

Comment:
why the check?

Reply:
`DefaultTppPassesOptions` could have some defaults.
It doesn't right now and probably shouldn't but it's not a bad practice.

Reply:
We can't provide defaults for shuffle and tiling as they are dependent on the tile shapes. I have added a default value for num-outer-parallel. The check is added because we take the older path in case parallel-task-grid option is provided and not tile shapes.